fpm/src/fpm_command_line.f90

```diff
@@ -203,24 +239,34 @@ subroutine get_command_line_settings(cmd_settings)
 
         case default
 
-            call set_args(' ', help_fpm, version_text)
-            ! Note: will not get here if --version or --usage or --help is present on commandline
-            if(len_trim(cmdarg).eq.0)then
-                write(stderr,'(*(a))')'ERROR: missing subcommand'
+            call set_args(' --list F', help_fpm, version_text)
+            ! Note: will not get here if --version or --usage or --help
+            ! is present on commandline
+            help_text=help_usage
+            if(lget('list'))then
+               help_text=help_list_dash
+            elseif(len_trim(cmdarg).eq.0)then
+                write(stdout,'(*(a))')'Fortran Package Manager:'
+                write(stdout,'(*(a))')' '
+                write(stdout,'(*(a))')' subcommand may be one of'
+                call printhelp(help_list_nodash)
             else
-                write(stderr,'(*(a))')'ERROR: unknown subcommand [', trim(cmdarg), ']'
+                write(stderr,'(*(a))')'ERROR: unknown subcommand [', &
+                 & trim(cmdarg), ']'
             endif
-            help_text=[character(len=80) ::                                            &
-             'USAGE: fpm [ SUBCOMMAND [SUBCOMMAND_OPTIONS] ] | [|--help|--version] ', &
-             '       Enter "fpm --help" for more information                       ', &
-             '' ]
-            write(stderr,'(g0)')(trim(help_text(i)), i=1, size(help_text) )
+            call printhelp(help_text)
```

Comment:
I think `"USAGE: fpm [ SUBCOMMAND [SUBCOMMAND_OPTIONS] ] | [--help|--version]"` etc. should be printed before the subcommand list - it makes more sense when reading.